server/src/main/java/org/opensearch/search/backpressure/settings/SearchBackpressureSettings.java

```diff
@@ -56,85 +46,23 @@ private static class Defaults {
         Setting.Property.NodeScope
     );
 
-    /**
-     * Defines the percentage of tasks to cancel relative to the number of successful task completions.
-     * In other words, it is the number of tokens added to the bucket on each successful task completion.
-     */
-    private volatile double cancellationRatio;
-    public static final Setting<Double> SETTING_CANCELLATION_RATIO = Setting.doubleSetting(
-        "search_backpressure.cancellation_ratio",
-        Defaults.CANCELLATION_RATIO,
-        0.0,
-        1.0,
-        Setting.Property.Dynamic,
-        Setting.Property.NodeScope
-    );
-
-    /**
-     * Defines the number of tasks to cancel per unit time (in millis).
-     * In other words, it is the number of tokens added to the bucket each millisecond.
-     */
-    private volatile double cancellationRate;
-    public static final Setting<Double> SETTING_CANCELLATION_RATE = Setting.doubleSetting(
-        "search_backpressure.cancellation_rate",
-        Defaults.CANCELLATION_RATE,
-        0.0,
-        Setting.Property.Dynamic,
-        Setting.Property.NodeScope
-    );
-
-    /**
-     * Defines the maximum number of tasks that can be cancelled before being rate-limited.
-     */
-    private volatile double cancellationBurst;
-    public static final Setting<Double> SETTING_CANCELLATION_BURST = Setting.doubleSetting(
-        "search_backpressure.cancellation_burst",
-        Defaults.CANCELLATION_BURST,
-        1.0,
-        Setting.Property.Dynamic,
-        Setting.Property.NodeScope
-    );
```

Comment:
I believe the change here is to remove common rate limiting settings `search_backpressure.cancellation_xyz` and create separate ones for each task type. This might be slightly concerning from a BWC point of view.

@Bukhtawar What are your thoughts on this? Are there any recommendations for such changes?

Reply:
Please address/response to this concern. This appears to be a breaking change.

Reply:
@ketanv3 @nssuresh2007 I have addressed this in the latest commit. Please have a look.
Thanks.